tests/generic_types.rs

```diff
@@ -1,25 +1,55 @@
-use mockiato::mockable;
-use std::fmt::{self, Display};
+use mockiato::{mockable, partial_eq};
+use std::fmt::{self, Debug, Display};
+
+trait Foo {
+    type Output;
+}
 
 #[mockable]
-trait Greeter<T>
+trait Greeter<T, U, V>
```

Comment:
Add a test for how this behaves with lifetimes at the start and end of the generic declaration list. We want nice error messages.

Reply:
Already covered by [mockiato-compiletest/tests/ui/trait_decl_with_non_type_generics.rs](https://github.com/myelin-ai/mockiato/blob/generics-on-traits/crates/mockiato-compiletest/tests/ui/trait_decl_with_non_type_generics.rs). Improved in 467cdce.

Reply:
Lifetimes at the end of the declaration list are invalid anyways, so rustc complains about those already.